src/status_im/utils/mixpanel.cljs

```diff
@@ -1,35 +1,56 @@
 (ns status-im.utils.mixpanel
   (:require-macros [status-im.utils.slurp :as slurp])
-  (:require [cljs.reader :as reader]
+  (:require [cljs.core.async :as async]
+            [cljs.reader :as reader]
             [goog.crypt.base64 :as b64]
+            [taoensso.timbre :as log]
             [status-im.utils.build :as build]
             [status-im.utils.config :as config]
             [status-im.utils.datetime :as datetime]
             [status-im.utils.http :as http]
             [status-im.utils.platform :as platform]
-            [status-im.utils.types :as types]))
+            [status-im.utils.types :as types]
+            [status-im.react-native.js-dependencies :as rn-dependencies]))
 
 (def base-url "http://api.mixpanel.com/")
 (def base-track-url (str base-url "track/"))
 
 (defn encode [m]
   (b64/encodeString (types/clj->json m)))
 
-(defn- build-url [id label props]
-  (str base-track-url
-       "?data="
-       (encode {:event      label
-                :properties (merge
-                              {:token       config/mixpanel-token
-                               :distinct_id id
-                               :os          platform/os
-                               :os-version  platform/version
-                               :app-version build/version
-                               :time        (datetime/timestamp)}
-                              props)})))
+(defn- make-event [id label props]
+  {:event      label
+   :properties (merge
+                {:token       config/mixpanel-token
+                 :distinct_id id
+                 :os          platform/os
+                 :os-version  platform/version
+                 :app-version build/version
+                 :time        (datetime/timestamp)}
+                props)})
+
+(def ^:private pending-events-queue (async/chan 2000))
+
+(async/go-loop [accumulator []]
```

Comment:
Id'd rather have the loop only used when the event can't be sent. The happy path would then be the raw http outside async, if the event can't be sent it's added in a queue.

Reply:
Similarly we want to control the lifecycle of the loop (start/stop). Having an always running loop is problematic in the case of switching users for instance.

Reply:
Will do!

Reply:
what do you guys think about using `mount` or `integrant` to manage stateful components?

Reply:
https://github.com/tolitius/mount/blob/master/doc/clojurescript.md#managing-state-in-clojurescript
https://github.com/weavejester/integrant

Reply:
It's definitively something we should consider. Would help with startup sequence and probably testing too.